extensions/wikia/FirstContributions/api/ApiQueryFirstContributions.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * API module which returns list of users' first edits, including revision id, date and user name
+ * @ingroup API
+ */
+class ApiQueryFirstContributions extends ApiQueryBase {
+
+	public function __construct( ApiBase $query, $moduleName ) {
+		parent::__construct( $query, $moduleName, 'fc' );
+	}
+
+	public function execute() {
+		$params = $this->extractRequestParams();
+
+		$this->addTables( 'recentchanges' );
+
+		$this->addFields( 'rc_this_oldid as id' );
+		$this->addFields( 'rc_timestamp as timestamp' );
+		$this->addFields( 'rc_user_text as author' );
+
+		// only list edit or new article creation
+		$this->addWhere( 'rc_type < ' . RC_MOVE );
+		$this->addWhere( [ 'rc_bot' => 0 ] );
+
+		$this->addTimestampWhereRange( 'rc_timestamp', $params['dir'], $params['after'], $params['before'] );
+
+		$this->addOption( 'GROUP BY', 'rc_user' );
+		$this->addOption( 'HAVING', 'id = (SELECT min(rev_id) from revision where rev_user = rc_user)' );
+		$this->addOption( 'LIMIT', $params['limit'] + 1 );
+
+		/*
+		 SELECT rc_this_oldid as id, rc_timestamp as timestamp, rc_user_text as author
+		 FROM recentchanges
+		 WHERE rc_type < 2 AND rc_bot = 0
+		 GROUP BY rc_user
+		 HAVING id = (SELECT min(rev_id) from revision where rev_user = rc_user)
+		 ORDER BY rc_timestamp DESC
+		 LIMIT 10;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)